doc/source/rllib/rllib-rlmodule.rst

```diff
@@ -0,0 +1,571 @@
+            ...
+            def input_specs_exploration(self) -> SpecType:
+                # Enforce that input nested dict to exploration method has a key "obs"
+                return ["obs"]
+            
+            def output_specs_exploration(self) -> SpecType:
+                # Enforce that output nested dict from exploration method has a key 
+                # "action_dist"
+                return ["action_dist"]
+
+.. tabbed:: Nested Keys
+
+    .. code-block:: python
+
+        from ray.rllib.models.specs.typing import SpecType
+        
+        class DiscreteBCTorchModule(TorchRLModule):
+            ...
+            def input_specs_exploration(self) -> SpecType:
+                # Enforce that input nested dict to exploration method has a key "obs"
+                # and within that key, it has a key "global" and "local". There should 
+                # also be a key "action_mask"
+                return [("obs", "global"), ("obs", "local"), "action_mask"]
+
+
+
+.. tabbed:: TensorShape Spec
+
+    .. code-block:: python
+
+        from ray.rllib.models.specs.typing import SpecType
+        from ray.rllib.models.specs.torch_spec import TorchTensorSpec
+        
+        class DiscreteBCTorchModule(TorchRLModule):
+            ...
+            def input_specs_exploration(self) -> SpecType:
+                # Enforce that input nested dict to exploration method has a key "obs"
+                # and its value is a torch.Tensor with shape (b, h) where b is the
+                # batch size (determined at run-time) and h is the hidden size 
+                # (fixed at 10).
+                return {"obs": TorchTensorSpec("b, h", h=10)}
+
+
+
+.. tabbed:: Type Spec
+
+    .. code-block:: python
+
+        from ray.rllib.models.specs.typing import SpecType
+        import torch
+        
+        class DiscreteBCTorchModule(TorchRLModule):
+            ...
+            def output_specs_exploration(self) -> SpecType:
+                # Enforce that output nested dict from exploration method has a key
+                # "action_dist" and its value is a torch.distribution.Categorical
+                return {"action_dist": torch.distribution.Categorical}
+
+:py:class:`~ray.rllib.core.rl_module.rl_module.RLModule` has two methods for each forward method, totaling 6 methods that can be override to describe the specs of the input and output of each method:
+
+- :py:meth:`~ray.rllib.core.rl_module.rl_module.RLModule.input_specs_inference`
+- :py:meth:`~ray.rllib.core.rl_module.rl_module.RLModule.output_specs_inference`
+- :py:meth:`~ray.rllib.core.rl_module.rl_module.RLModule.input_specs_exploration`
+- :py:meth:`~ray.rllib.core.rl_module.rl_module.RLModule.output_specs_exploration`
+- :py:meth:`~ray.rllib.core.rl_module.rl_module.RLModule.input_specs_train`
+- :py:meth:`~ray.rllib.core.rl_module.rl_module.RLModule.output_specs_train`
+
+To learn more about this see the documentation of `SpecType`.
```

Comment:
```suggestion
To learn more, see the `SpecType` documentation.
```

Reply:
Suggest linking to the `SpecType` doc.

Reply:
It should be written in a separate PR. But temporary should be ok.